source/usecases/usecase2/nodes.h5.txt

```diff
@@ -265,6 +286,18 @@ GROUP "/" {
       GROUP "VirtualPopA" {
          GROUP "0" {
             GROUP "@library" {
+               DATASET "model_template" {
+                  DATATYPE  H5T_STRING {
+                     STRSIZE H5T_VARIABLE;
+                     STRPAD H5T_STR_NULLTERM;
+                     CSET H5T_CSET_UTF8;
+                     CTYPE H5T_C_S1;
+                  }
+                  DATASPACE  SIMPLE { ( 1 ) / ( 1 ) }
+                  DATA {
+                     ""
```

Comment:
in the spec we say it is NULL . so we mean "" actually ?

Reply:
That is what I don't know. The spec doesn't explicitly say what it means, so there's room for interpretation. The dtype is `utf-8`, python does not have `NULL`, so I guess the `str` equivalent is an empty string. Or do we want it to actually be `"NULL"` as a `str`?

Reply:
probably "" is best. I don't recall if there is something that means "unset" in hdf5

Reply:
There's the [empty dataset](https://docs.h5py.org/en/stable/high/dataset.html#creating-and-reading-empty-or-null-datasets-and-attributes). Should we be using that instead?

Reply:
I think `""` makes the most sense.